zulipterminal/ui_tools/messages.py

```diff
@@ -729,6 +730,36 @@ def main_view(self) -> List[Any]:
                 "/me", f"<strong>{self.message['sender_full_name']}</strong>", 1
             )
 
+        # If message contains submessages (like polls, todo), process them
+        # and update the message content.
+        if self.message["submessages"]:
+            try:
+                first_submessage_content = json.loads(
+                    self.message["submessages"][0]["content"]
+                )
+            except (json.JSONDecodeError, TypeError):
+                first_submessage_content = {}
+
+            if (
+                "widget_type" in first_submessage_content
+                and first_submessage_content["widget_type"] == "poll"
+            ):
+                question, votes_for_option = self.process_poll_data(
+                    self.message["submessages"]
+                )
```

Comment:
Extending Neil's comment, since this function is to be added in the model file, the code patch for `process_poll_data` and its test can be shifted to a separate commit which comes before this commit.